### PR TITLE
Proposed pom cleanup, with missing api dependency

### PR DESCRIPTION
**Pull request by niloc132**
Adds missing dependency to maven metadata, removes an unnecessary tag, and adjusts a popular URL to use https.

### CODE REVIEW

**Comment by niloc132** [comment]
Presently using jspecify 0.3.0 based on https://github.com/google/j2cl/blob/1fa454a14f26274efbb35a81463fbafea2e89666/build_defs/workspace.bzl#L67-L72 as that seems to be latest for j2cl (and so this repo), but the differences to 1.0.0 seem very small and projects can manage that version difference if desired.

Verified this change by installing it locally and updating an existing project to use this, elemental2 1.2.3, and GWT 2.12.1 (to be released tomorrow, will resolve the issue mentioned in https://github.com/google/jsinterop-base/issues/19#issuecomment-2442674784).

Technically only the jspecify dependency is required, but just looked quickly for other cleanup as long as was in the file.

File: maven/pom-base.xml [hunk 1]
```diff
@@ -12,7 +12,6 @@
   <groupId>__GROUP_ID__</groupId>
   <artifactId>__ARTIFACT_ID__</artifactId>
   <version>__VERSION__</version>
-  <packaging>jar</packaging>
```
Comment (jDramaix) [inline maven/pom-base.xml:15]: OOC why did you remove this ? The default value is `pom` right ?

Comment (niloc132) [reply maven/pom-base.xml:15]: No, the default is jar, you need to specify it for pom/war/etc, but it need not be specified for jar.

https://maven.apache.org/pom.html#Packaging

> When no packaging is declared, Maven assumes the packaging is the default: `jar`.

Comment (jDramaix) [reply maven/pom-base.xml:15]: and you don't inherit the one from the parent pom ?

Comment (niloc132) [reply maven/pom-base.xml:15]: Correct. That wouldn't make any sense, since all poms, even without an explicit parent, have a parent pom that defines plugin versions, etc, and no project could ever take advantage of that default if it used a parent pom (if packaging isn't "pom", you can't inherit from it). https://maven.apache.org/pom.html#inheritance

Comment (niloc132) [reply maven/pom-base.xml:15]: If you'd rather I revert this line, I'm happy to do so, there's no problem with being explicit about defaults. It just feels a little like calling object.tostring() when already concatenating with other strings, it isn't wrong, its just verbose.

Comment (jDramaix) [reply maven/pom-base.xml:15]: nope I'm fine, I need to do that for the other projects in order to be consistent

**Review by zbynek — APPROVED** [review]
